### DISCUSSION TITLE
PandasGUI on Google Colab?

### DISCUSSION THREAD

**Discussion by yerbby** [adamerose/pandasgui#197] [Q&A] [answered]
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
[<ipython-input-2-f7d5d62bc021>](https://localhost:8080/#) in <module>()
      1 import pandas as pd
----> 2 from pandasgui import show
      3 df = pd.DataFrame({'a':[1,2,3], 'b':[4,5,6], 'c':[7,8,9]})
      4 show(df)

5 frames
[/usr/local/lib/python3.7/dist-packages/pynput/_util/__init__.py](https://localhost:8080/#) in backend(package)
     80                 ' * {}'.format(s)
     81                 for s in resolutions))
---> 82             if resolutions else '')
     83 
     84 

ImportError: this platform is not supported: ('failed to acquire X connection: Bad display name ""', DisplayNameError(''))

Try one of the following resolutions:

 * Please make sure that you have an X server running, and that the DISPLAY environment variable is set correctly

**Comment by adamerose** [comment, marked as answer]
Unfortunately this is currently not possible, since PandasGUI is built using PyQt5 which does not support Google Colab. I have ideas about eventually rebuilding the UI using React which will support Jupyeter Notebook embedding and Google Colab, but that is a large undertaking and I'm considering ways to monetize the project if I decide to do such a rewrite.
